src/rust/engine/src/scheduler.rs

```diff
@@ -89,22 +104,48 @@ impl Scheduler {
     field: Field,
     field_types: Vec<TypeId>
   ) {
+    // find the edges for the root in the rule graph
```

Comment:
I'm sure we'll eventually do `rustfmt` or something. But in the meantime, let's keep the same comment style that we use for python (capitalized sentences ending in periods). It has... grown on me somehow =/